src/components/expansion-panel/expansion-panel.ts

```diff
@@ -0,0 +1,239 @@
+import { LitElement, html } from 'lit';
+import { property, query } from 'lit/decorators.js';
+import { Constructor } from '../common/mixins/constructor';
+import { EventEmitterMixin } from '../common/mixins/event-emitter';
+import { themes } from '../../theming/theming-decorator.js';
+import { styles } from './themes/light/expansion-panel.base.css.js';
+import { styles as bootstrap } from './themes/light/expansion-panel.bootstrap.css.js';
+import { styles as fluent } from './themes/light/expansion-panel.fluent.css.js';
+import { styles as indigo } from './themes/light/expansion-panel.indigo.css.js';
+
+let NEXT_ID = 0;
+export interface IgcExpansionPanelComponentEventMap {
+  igcOpening: CustomEvent<any>;
+  igcOpened: CustomEvent<any>;
+  igcClosing: CustomEvent<any>;
+  igcClosed: CustomEvent<any>;
+}
+
+const TABBABLE_SELECTORS =
+  'a[href], button, input, textarea, select, details, [tabindex]:not([tabindex="-1"])';
+
+/**
+ * The Expansion Panel Component provides a way to display information in a toggleable way -
+ * compact summary view containing title and description and expanded detail view containing
+ * additional content to the summary header.
+ *
+ * @element igc-expansion-panel
+ *
+ * @slot title - renders the title of the panel's header
+ * @slot subTitle - renders the subtitle of the panel's header
+ * @slot indicator - renders the expand/collapsed indicator
+ *
+ * @csspart header
+ * @csspart title
+ * @csspart subTitle
+ * @csspart indicator
+ * @csspart content
+ */
+@themes({ bootstrap, fluent, indigo })
+export default class IgcExpansionPanelComponent extends EventEmitterMixin<
+  IgcExpansionPanelComponentEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  public static readonly tagName = 'igc-expansion-panel';
+  public static styles = styles;
+
+  /** Indicates whether the contents of the control should be visible. */
+  @property({ reflect: true, type: Boolean })
+  public open = false;
+
+  /** Get/Set whether the expansion panel is disabled. Disabled panels are ignored for user interactions. */
+  @property({ reflect: true, type: Boolean })
+  public disabled = false;
+
+  /** The indicator alignment of the expansion panel. */
+  @property({ reflect: true, attribute: 'indicator-alignment' })
+  public indicatorAlignment: 'start' | 'end' = 'start';
+
+  @query('[part~="header"]', true)
+  protected panelHeader!: HTMLElement;
+
+  private panelId!: string;
+
+  constructor() {
+    super();
+  }
+
+  public override connectedCallback() {
+    super.connectedCallback();
+    const id = this.getAttribute('id');
+    this.panelId! = id ? id : 'igc-expansion-panel-' + ++NEXT_ID;
+  }
+
+  private handleClicked(event: Event) {
+    if (this.disabled) {
+      event.preventDefault();
+      return;
+    }
```

Comment:
**Suggestion**: IMO this will be unnecessary if you apply `pointer-events: none` on the header part when the control is disabled.